lib/circuits/baselineDocument.zok

```diff
@@ -0,0 +1,110 @@
+import "utils/pack/pack128" as pack128
+import "utils/pack/pack256" as pack256
+import "utils/pack/nonStrictUnpack256" as unpack256
+import "ecc/edwardsCompress" as edwardsCompress
+import "./verifyEddsa" as verifyEddsa
+import "ecc/babyjubjubParams" as context
+
+import "./hashes/sha256/sha256of2x256" as sha256of2x256
+import "./hashes/sha256/sha256of5x128" as sha256of5x128
+import "./hashes/sha256/sha256of7x128" as sha256of7x128
+import "./hashes/sha256/sha256of8x128" as sha256of8x128
+import "./packing/unpack2x128To256x1.zok" as unpack2x128To256x1
+
+import "./merkle-tree/calc-root-from-siblingPath-sha256-height4" as calculateRoot
+
+struct EdDSASignature {
+  field[2] R
+  field S
+}
+
+
+//structure of agreement:
+// { inputs {}, commitments [linked list], root, calldata}
+
+struct DocumentInputs {
+  field[2] publicKeyOfBuyer
```

Comment:
Should we remove all references to `buyer` and `supplier` and replace with something more general such as `party1/party2` or `sender/receipient`?

Reply:
Yes.

Reply:
+1 for removing `buyer` `supplier` but would caution against referentially dependent names like `sender/recipient` if the same entity can be both a sender and recipient in a single baseline transaction.

Reply:
I was planning to use `party` or `participant`. Any preference?